### PR TITLE
rpl: switch to xtimer

### PR DESCRIPTION
**Pull request by cgundogan**
This PR replaces the vtimer with xtimer in RPL.

**EDIT:**
Rationale: without this switch, strange `enableIRQ` warnings occure on native when initializing rpl.
Furthermore, this is an alternative to #3775, without dropping existing features.

~~depends on: #3776~~

### CODE REVIEW

**Comment by kaspar030** [comment]
See #3776: we probably shouldn't merge into master, but into switch_to_xtimer.

**Comment by cgundogan** [comment]
rebased and squashed

File: sys/net/gnrc/routing/rpl/gnrc_rpl.c [hunk 1]
```diff
@@ -217,44 +217,41 @@ static void *_event_loop(void *args)
...
+                * SEC_IN_USEC)) {
                 gnrc_rpl_dodag_t *dodag = parent->dodag;
                 gnrc_rpl_parent_remove(parent);
                 gnrc_rpl_parent_update(dodag, NULL);
                 continue;
             }
-            else if (((signed)(parent->lifetime.seconds - now.seconds) <=
-                        GNRC_RPL_LIFETIME_UPDATE_STEP * 2)) {
+            else if ((int64_t)(parent->lifetime - now) <= (int64_t) (GNRC_RPL_LIFETIME_UPDATE_STEP
+                     * SEC_IN_USEC * 2)) {
```
Comment (OlegHahm) [inline sys/net/gnrc/routing/rpl/gnrc_rpl.c]: minor: I think a line break after the comparator would be more readable.

File: sys/shell/commands/sc_gnrc_rpl.c [hunk 2]
```diff
@@ -313,25 +314,30 @@ int _gnrc_rpl_dodag_show(void)
...
+            ti = (((uint64_t) dodag->trickle.msg_interval_timer.long_target << 32)
+                    | dodag->trickle.msg_interval_timer.target) - xnow;
+            ti = (int64_t) ti < 0 ? 0 : ti / SEC_IN_USEC;
+
+            cleanup = dodag->cleanup_timer.target - xtimer_now();
+            cleanup = (int32_t) cleanup < 0 ? 0 : cleanup / SEC_IN_USEC;
+
             printf("\tdodag [%s | R: %d | CL: %" PRIu32 "s | \
-TR(I=[%d,%d], k=%d, c=%d, TC=%" PRIu32 "s, TI=%" PRIu32 "s)]\n",
+TR(I=[%d,%d], k=%d, c=%d, TC=%" PRIu64 "s, TI=%" PRIu64 "s)]\n",
```
Comment (OlegHahm) [inline sys/shell/commands/sc_gnrc_rpl.c]: Can you fix the indentation while you're at it?

File: sys/shell/commands/sc_gnrc_rpl.c [hunk 3]
```diff
@@ -313,25 +314,30 @@ int _gnrc_rpl_dodag_show(void)
...
                     ipv6_addr_to_str(addr_str, &dodag->dodag_id, sizeof(addr_str)),
-                    dodag->my_rank, ((int32_t) cleanup.seconds) > 0 ? cleanup.seconds : 0,
+                    dodag->my_rank, ((int32_t) cleanup < 0 ? 0 : cleanup/SEC_IN_USEC),
```
Comment (OlegHahm) [inline sys/shell/commands/sc_gnrc_rpl.c]: parens

**Comment by OlegHahm** [comment]
Apart from minor comments: ACK

**Comment by cgundogan** [comment]
addressed comments of @OlegHahm (amended)

**Comment by cgundogan** [comment]
@kaspar030 do I need to rebase this PR on one of your branches, or is everything fine this way?

**Comment by kaspar030** [comment]
@cgundogan everything fine!

- merged this into #3525 (switch_to_xtimer)

**Pull request closed without merging**